pyteal/compiler/flatten.py

```diff
@@ -123,7 +143,16 @@ def flattenSubroutines(
             if isinstance(stmt, TealLabel):
                 stmt.getLabelRef().addPrefix(labelPrefix)
 
-        combinedOps.append(TealLabel(None, LabelReference(label), comment))
+        # this is needed for source map generation
+        dexpr: SubroutineDeclaration | None
+        try:
+            dexpr = subroutine.declarations.get_declaration_by_option(
```

Comment:
nit: I think you can shrink the code by directly call
```suggestion
            dexpr = subroutine.get_declaration_by_option(
```

Reply:
a27396b6c8b17a96b538ce1fee2e83cb3f1a63d6

Reply:
looking good, resolving